source/slang/hlsl.meta.slang

```diff
@@ -195,6 +195,14 @@ struct __ShapeBuffer : __ITextureShape
     static const int dimensions = 1;
     static const int planeDimensions = 1;
 }
+__magic_type(TextureShapeSubpassType)
```

Comment:
Does this have to be in hlsl.meta.slang?

Any shape we add here is going to contirbute to a gigantic cross-product space of all texture type combinations and it is going to increase the size of the stdlib, decreasing the startup performance by noticeable amount, so we need to decide if it is worth it.

Reply:
I will look to see if this type needs to be here

Reply:
in order to combat this issue I changed the texture inheritance hierarchy to minimize the scope of each generic (these changes reflect on the `__ITextureImpl` as well

`__ITextureShape` ->
    -> `__ITextureShapeUnique `
         -> `Subpass`
    -> `__ITextureShapeTraditional`
         -> `__ITextureShape1D2D3D`
              -> 1D, 2D etc...
         -> Buffer, Cube, etc....